packages/lit-dev-content/site/articles/article/lit-with-tailwind.md

```diff
@@ -0,0 +1,228 @@
+the associated CSS at build time (via postcss).
+
+This conflicts with how web components work, since each web component has its
+own natively scoped stylesheet rather than inherting any global styles. This
+is why Tailwind will not be much use to us without further setup.
+
+## Overview of solution
+
+To solve the gaps tailwind comes with, we will:
+
+- Inject tailwind styles into regular CSS files
+- Use [import attributes](https://github.com/tc39/proposal-import-attributes)
+to import those CSS files
+- Use [esbuild](https://github.com/evanw/esbuild) to pull those CSS files
+into the bundle or same output directory
+
+## Initial setup
+
+To begin, we need the following dependencies for our build:
+
+- `esbuild` for bundling our code and carry our CSS files across (via imports)
+- `postcss` for injecting tailwind styles
+- `tailwindcss` for tailwind itself (postcss plugin)
+
+We can install these like so:
+
+```sh
+npm install -D esbuild postcss postcss-cli tailwindcss
+```
+
+In our case, we are going to use `postcss-cli` for processing our CSS files. If
+you use rollup or another bundler already, you may be able to use a postcss
+plugin instead.
+
+## Using CSS imports
+
+When creating components, we want to use
+[import attributes](https://github.com/tc39/proposal-import-attributes) to
+import our CSS files rather than embedding CSS in our sources.
+
+**However, at the time of writing this article, the new `with` syntax is not
+fully supported**. Temporarily, we still have to use the old `assert` syntax,
+until esbuild implements full support for the new standard.
+
+We can do this like so:
+
+```ts
+import myElementStyles from './my-element.css' assert {type: 'css'}
+
+class MyElement extends LitElement {
+  static styles = myElementStyles;
+
+  render() {
+    return html`
+      <span class="italic">I am italic tailwind text</span>
+    `;
+  }
+}
+```
+
+As you can see, we want to our `my-element.css` file to contain tailwind
+mixins, and we want to use the resulting classes in our element's `render`
+method.
+
+The CSS file (`my-element.css`) would look like this:
+
+```css
+@tailwind base;
+@tailwind utilities;
+
+/* other custom CSS here */
+```
+
+We will then use tailwind to replace those mixins with CSS, including only
+the styles our `render` method referenced.
+
+## Configuring tailwind and postcss
+
+Before we run our build, we need to configure tailwind and postcss.
+
+In our `tailwind.config.js`, we can write:
+
+```ts
+/** @type {import('tailwindcss').Config} */
+export default {
+  content: [
+    './src/**/*.ts'
+  ],
+  theme: {
+    extend: {},
+  },
+  plugins: [],
+}
+```
+
+In our case, our sources are TypeScript, so we have the initial `@type` comment
+to give us auto-completion for the object's properties.
+
+The rest of this file is straight forward. The important part is `content`,
+specifying where our sources are so tailwind can detect which classes we have
+used.

```

Comment:
I'm not a huge fan of saying something is "straight forward" just in case it's not straight forward for some readers.

```suggestion
Tailwind [configuration](https://tailwindcss.com/docs/configuration) can be modified to fit your use case. The important part is for `content` to specify where our component sources are so tailwind can detect which classes we have used.
```